packages/core/src/asset/ResourceManager.ts

```diff
@@ -182,17 +182,48 @@ export class ResourceManager {
   /**
    * @internal
    */
-  _onSubAssetSuccess<T>(assetURL: string, value: T): void {
-    this._subAssetPromiseCallbacks[assetURL]?.resolve(value);
-    delete this._subAssetPromiseCallbacks[assetURL];
+  _onSubAssetSuccess<T>(assetBaseURL: string, assetSubPath: string, value: T): void {
+    const subPromise = this._subAssetPromiseCallbacks[assetBaseURL]?.[assetSubPath];
+    if (subPromise) {
+      subPromise.resolve(value);
+    } else {
+      (this._subAssetPromiseCallbacks[assetBaseURL] ||= {})[assetSubPath] = {
+        resolve: value
```

Comment:
`subPromise.resolve(value)` will call,`  resolve: value` just assign，Are you writing unspoken rules again?